### PR TITLE
Add br boot settings

### PR DESCRIPTION
**Pull request by ahreehong**
*Description of changes:*
Add support for configuring boot.kernel-parameters for bottlerocket nodes.

*Testing (if applicable):*
Manually tested by creating clusters with vsphere and snow providers

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/clusterapi/bottlerocket.go [hunk 1]
```diff
@@ -120,23 +123,54 @@ func SetBottlerocketControlContainerImageInKubeadmConfigTemplate(kct *bootstrapv
 
 // SetBottlerocketHostConfigInKubeadmControlPlane sets bottlerocket specific kernel settings in kubeadmControlPlane.
 func SetBottlerocketHostConfigInKubeadmControlPlane(kcp *controlplanev1.KubeadmControlPlane, hostOSConfig *anywherev1.HostOSConfiguration) {
-	if hostOSConfig == nil || hostOSConfig.BottlerocketConfiguration == nil || hostOSConfig.BottlerocketConfiguration.Kernel == nil ||
-		hostOSConfig.BottlerocketConfiguration.Kernel.SysctlSettings == nil {
+	if hostOSConfig == nil || hostOSConfig.BottlerocketConfiguration == nil || (hostOSConfig.BottlerocketConfiguration.Kernel == nil &&
```
Comment (jiayiwang7) [inline pkg/clusterapi/bottlerocket.go:126]: nit: can we do something like

```suggestion
	if hostOSConfig == nil || hostOSConfig.BottlerocketConfiguration == nil {
	return nil
	}
```

then update `hostConfig()` to include the boot settings, use it in all below, to remove the duplicate code

File: pkg/api/v1alpha1/hostosconfig_types.go [hunk 2]
```diff
@@ -26,4 +26,7 @@ type BottlerocketConfiguration struct {
 
 	// Kernel defines the kernel settings for bottlerocket.
 	Kernel *v1beta1.BottlerocketKernelSettings `json:"kernel,omitempty"`
+
+	// BootSettings defines the boot settings for bottlerocket.
+	BootSettings map[string][]string `json:"bootSettings,omitempty"`
```
[the remark this answers is not in the document]
Comment (ahreehong) [reply pkg/api/v1alpha1/hostosconfig_types.go:31]: I forgot to change the bootsettings field to inherit from capi when I updated the kernel field. I'll update it to inherit from capi so it's more consistent

**Pull request merged**